### PR TITLE
Improve cunumeric.diagonal code coverage

### PR DESCRIPTION
**Pull request by yimoj**

### CODE REVIEW

File: tests/integration/test_index_routines.py [hunk 1]
```diff
@@ -412,6 +427,25 @@ def test_axes_none(self):
...
 
+    def test_scalar_axes(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axes=(0,))
+
+    def test_duplicate_axes(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axis1=1, axes=(0, 1))
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axis1=1, axis2=0, axes=(0, 1))
```
Comment (bryevdv) [inline tests/integration/test_index_routines.py:438]: Let's follow the `expected_exc` pattern established in other recent tests

File: tests/integration/test_index_routines.py [hunk 2]
```diff
@@ -412,6 +427,25 @@ def test_axes_none(self):
...
+
+    def test_extra_axes(self):
+        axes = num.arange(self.a.ndim + 1, dtype=int)
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, axes=axes)
+
+    def test_n_axes_offset(self):
+        with pytest.raises(ValueError):
+            num.diagonal(self.a, offset=1, axes=(2, 1, 0))
```
Comment (bryevdv) [inline tests/integration/test_index_routines.py:447]: For these three tests that only assert a cunumeric exception, what is the numpy behavior? If numpy doesn't raise the same error then these tests seem like they ought to be xfails instead, unless this is an intended divergence from numpy.

Comment (yimoj) [reply tests/integration/test_index_routines.py:447]: NumPy doesn't have "axes" arg at all.

Comment (yimoj) [reply tests/integration/test_index_routines.py:446]: @bryevdv any more comments on this one?

Comment (bryevdv) [reply tests/integration/test_index_routines.py:446]: > NumPy doesn't have "axes" arg at all.

In that case maybe just add a comment to that effect. We could also add `@pytest.mark.diff` that was introduced in another recent PR (just a way to quickly identify tests that codify an intended/expected difference from numpy)

**Comment by yimoj** [comment]
Had to force push to fix the commit history after code rebase, actual change is just the last commit.

**Pull request merged**